snap7/client_async.py

```diff
@@ -3,22 +3,28 @@
 """
 import asyncio
 import logging
-from ctypes import c_int, byref, c_byte
+from ctypes import c_int, byref
 
-import snap7
 from snap7.common import check_error
-from snap7.types import buffer_type, buffer_size
+import snap7
 from .client import Client
+from .types import buffer_type, buffer_size
 
 logger = logging.getLogger(__name__)
 
 
 def error_wrap(func):
     """Parses a s7 error code returned the decorated function."""
 
-    def f(*args, **kw):
-        code = func(*args, **kw)
-        check_error(code, context="client")
+    if asyncio.iscoroutinefunction(func):
```

Comment:
i think i would make an explicit async error wrap function and use that for async functions, since you already know it is going to be a async function while programming